core/templates/dev/head/filters.js

```diff
@@ -404,3 +404,34 @@ oppia.filter('stripFormatting', [function() {
     return strippedText;
   };
 }]);
+
+/* The following filter replaces each RTE element occurrence in the input html
+   by its corresponding name in square brackets and returns a string
+   which contains the name in the same location as in the input html.
+   eg: <p>Sample1 <oppia-noninteractive-math></oppia-noninteractive-math>
+        Sample2 </p>
+   will give as output: Sample1 [Math] Sample2 */
+oppia.filter('formatRtePreview', ['$filter', function($filter) {
+  return function(html) {
+    html = html.replace(/&nbsp;/ig, ' ');
+    html = html.replace(/&quot;/ig, '');
+    //Replace all html tags other than <oppia-noninteractive-**> ones to ''
+    html = html.replace(/<(?!oppia-noninteractive\s*?)[^>]+>/g, '');
+    html = html.split(/[<>]+/);
+    var formattedOutput = '';
+    for (var i = 0; i < html.length; i++) {
```

Comment:
Instead of a manual loop, can you use html.replace() similar to https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/expressions/ExpressionInterpolationService.js#L45 ? Or is that trickier?

Reply:
(To be clear, I was thinking that the regex to look for when replacing is something that matches `<oppia-noninteractive-* ...></oppia-noninteractive-*>`.)

Reply:
That may be trickier as all I am using is the text just after <oppia-noninteractive- but without the attributes following that, can a disjointed string like that be replaced using string.replace()?

Reply:
I think so? Should be possible to represent it with a regex and a capture group, where the "..." stuff is basically "any character that's not '>'"

Something like `<oppia-noninteractive-([a-z]+) [^>]*></oppia-noninteractive-([a-z]+)>` maybe?

Reply:
Done, used a function in the replace string part of string.replace() to replace the entire tag with the name of the rte component